### PR TITLE
sigmahq title first draft

### PR DESCRIPTION
**Pull request by frack113**
Adds help for writing rule titles in SigmaHQ

### CODE REVIEW

File: sigmahq/sigmahq_title_rule.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+Bear in mind that the title is the first thing the operator will see.
+It should therefore be used as a clue to guide the consideration of the alert.
+
+The title and level of the rule must be consistent
+
+
+## Structure
+
+Title can be split with "-" : `Prefix - Description - Sufix`
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:26]:
```suggestion
Titles can be split with "-" : `Prefix - Description - Sufix`

```

File: sigmahq/sigmahq_title_rule.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+### Prefix
+It is Optional.
+It is used to give a category or type of malware, an APT 
+
+Commun wordding:
+- Hack Tool
+- Lolbin (lolbas)
+- PUA
+- Remote Access Tool
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:36]:
```suggestion
- HackTool
- PUA
- Remote Access Tool

```

File: sigmahq/sigmahq_title_rule.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+## Structure
+
+Title can be split with "-" : `Prefix - Description - Sufix`
+
+### Prefix
+It is Optional.
+It is used to give a category or type of malware, an APT 
+
+Commun wordding:
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:32]:
```suggestion
Examples:

```

File: sigmahq/sigmahq_title_rule.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+- Remote Access Tool
+
+Specific wording example:
+- "ATP27 - "
+- "ATP29 - "
+- "UNC2452 - "
+- "UNC4841 - "
+
+### Suffix
+It is Optional.
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:45]:
```suggestion
### Suffix (Optional)

```

File: sigmahq/sigmahq_title_rule.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+It is used to differentiate the same detection but on a different logsource.
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:46]:
```suggestion
Sometimes the detection are duplicated across different log-source with little changes to their logic. This is common in the case of Process Creation rules targeting the PowerShell process and rules using ScriptBlockText to check for the same. A suffix in this case will be used to offer such distinction.

```

File: sigmahq/sigmahq_title_rule.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+Exemple:
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:48]:
```suggestion
Example:


```

File: sigmahq/sigmahq_title_rule.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+
+`informational` and `low` are not intended for everyday use and are subject to many false positives.
+The title should therefore be general.
+
+Example : `Net.exe Execution` 
+
+`medium` rules can have false positives and requires further analysis
+Wording :
+- "Suspicious "
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:78]:
```suggestion
- "Potential "

```

File: sigmahq/sigmahq_title_rule.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+
+`high` rules requires a prompt review
+Wording :
+- "Potential "
```
Comment (nasbench) [inline sigmahq/sigmahq_title_rule.md:83]:
```suggestion
- "Suspicious "

```

**Comment by nasbench** [comment]
I did a first quick review. @phantinuss care to take a look to maybe add/remove things.

We want to use this as a general guidelines to add to the SIGMA convention.

File: sigmahq/sigmahq_title_rule.md [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+- "Renamed xxx Execution"
+- "UAC Bypass Using ..."
+
+
+`informational` and `low` are not intended for everyday use and are subject to many false positives.
+The title should therefore be general.
+
+Example : `Net.exe Execution` 
+
+`medium` rules can have false positives and requires further analysis
```
Comment (phantinuss) [inline sigmahq/sigmahq_title_rule.md:83]:
```suggestion
`medium` rules can have environment dependent false positives and require a tuning/evaluation phase before deploying to production environments.

```

File: sigmahq/sigmahq_title_rule.md [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+
+
+`high` rules requires a prompt review
+Wording :
+- "Suspicious "
+
+
+
+`critical` rules should be reviewed immediately
```
Comment (phantinuss) [inline sigmahq/sigmahq_title_rule.md:95]:
```suggestion
`critical` rules should be reviewed with the highest priority.

```

**Comment by phantinuss** [comment]
I like the idea of this and I think it will be a useful resource.

File: sigmahq/sigmahq_title_rule.md [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+The title must therefore be precise.
```
[the remark this answers is not in the document]
Comment (phantinuss) [reply sigmahq/sigmahq_title_rule.md:96]: maybe? 
```suggestion
`critical` rules should be reviewed immediately
The title must therefore be precise and indicate the threat.

Keywords used to indicate this:
- "Malware"
- "Exploit"
- "... Attempt"
- "<Threat Actor> Activity"
```

[the remark this answers is not in the document]
Comment (phantinuss) [reply sigmahq/sigmahq_title_rule.md:96]: But as I said. I wouldn't want to enforce anything there. First we would need to find good keywords to use.

[the remark this answers is not in the document]
Comment (frack113) [reply sigmahq/sigmahq_title_rule.md:96]: I wanted to remain as open as possible.
From a syntax perspective "Use of 7zip" and "Use of Brute Ratel" are the same.
Operator have the title and the level to differentiate them.

 "High-Risk Use of Brute Ratel"  , but why high for a critical 😄 
threatening, vicious, menacing ...

I don't see any solution

File: sigmahq/sigmahq_title_rule.md [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell
+title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell Module
+title: Invoke-Obfuscation Obfuscated IEX Invocation - Security
+title: Invoke-Obfuscation Obfuscated IEX Invocation - System
+```
+
+### Description
+
+The point here is to explain the alert in a meaningful way.
+The title does not need to use the terms "Detect", "Detection" or "Detection".
```
[the remark this answers is not in the document]
Comment (nasbench) [reply sigmahq/sigmahq_title_rule.md:73]: The description here refers to the structure defined above `Titles can be split with "-" : Prefix - Description - Sufix`

Maybe we can find a better word than description to not get it confused with the "real" description field.

File: sigmahq/sigmahq_title_rule.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (nasbench) [reply sigmahq/sigmahq_title_rule.md]: I think the intention is clear, we don't enforce it to not be cumbersome to the author but we'll review anyway so just having it as its now is good guidance, especially when the author will look at some example of critical rules. They should get the idea.

File: sigmahq/sigmahq_title_rule.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (frack113) [reply sigmahq/sigmahq_title_rule.md:73]: `Titles can be split with "-" : Prefix - Main Text - Sufix` ?

File: sigmahq/sigmahq_title_rule.md [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+
+```yaml
+title: Invoke-Obfuscation Obfuscated IEX Invocation
+title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell
+title: Invoke-Obfuscation Obfuscated IEX Invocation - PowerShell Module
+title: Invoke-Obfuscation Obfuscated IEX Invocation - Security
+title: Invoke-Obfuscation Obfuscated IEX Invocation - System
+```
+
+### Description
```
Comment (phantinuss) [inline sigmahq/sigmahq_title_rule.md:69]:
```suggestion
### Main Title

```

File: sigmahq/sigmahq_title_rule.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (phantinuss) [reply sigmahq/sigmahq_title_rule.md]: What's with the code suggestion of this thread? I think it makes sense to give examples, doesn't it?

**Review by nasbench — APPROVED** [review]

File: sigmahq/sigmahq_title_rule.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (nasbench) [reply sigmahq/sigmahq_title_rule.md]: Applied the suggestion

**Review by phantinuss — APPROVED** [review]

**Pull request merged**